massa-final-state/src/final_state.rs

```diff
@@ -579,17 +580,22 @@ impl FinalState {
     ///
     /// Panics if the new slot is not the one coming just after the current one.
     pub fn finalize(&mut self, slot: Slot, changes: StateChanges) {
-        let cur_slot = self.db.read().get_change_id().expect(CHANGE_ID_DESER_ERROR);
+        self._finalize(slot, changes).unwrap()
```

Comment:
Won't this unwrap interfere with error handling? (e.g. if the pos_state apply changes throws an error, would that be picked up?)

Reply:
_finalize returns an error on pos_state.apply_changes and the previous code did an expect so I would say that it's ok no?

Reply:
Yes, I don't know why but I thought the error message would not show up :)